notebooks/02_model/rbm_deep_dive.ipynb

```diff
@@ -0,0 +1,1449 @@
+    "\n",
+    "Below, we first load the different movielens data in pandas dataframes, explain how the user/affinity matrix is built and how the train/test set is generated. As this procedure is common to all the datasets we will consider, we explain it in details only for the 1 milion dataset.  \n",
+    "\n",
+    "We start with downloading the different dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 58,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# MovieLens data have different data-format for each size of dataset \n",
+    "data_header = None\n",
+    "\n",
+    "def get_data( MOVIELENS_DATA_SIZE):\n",
+    "\n",
+    "    if MOVIELENS_DATA_SIZE == '100k':\n",
+    "        separator = '\\t'\n",
+    "        data_name = 'u.data'\n",
+    "        data_folder = 'ml-100k'\n",
+    "\n",
+    "    elif MOVIELENS_DATA_SIZE == '1m':\n",
+    "        separator = '::'\n",
+    "        data_name = 'ratings.dat'\n",
+    "        data_folder = 'ml-1m'\n",
+    "\n",
+    "    elif MOVIELENS_DATA_SIZE == '10m':\n",
+    "        separator = '::'\n",
+    "        data_name = 'ratings.dat'\n",
+    "        data_folder = 'ml-10M100K'\n",
+    "\n",
+    "    elif MOVIELENS_DATA_SIZE == '20m':\n",
+    "        separator = ','\n",
+    "        data_name = 'ratings.csv'\n",
+    "        data_folder = 'ml-20m'\n",
+    "        data_header = 0\n",
+    "    \n",
+    "    else:\n",
+    "        raise ValueError('Invalid data size. Should be one of {100k, 1m, 10m, or 20m}') \n",
+    "\n",
+    "    # Download dataset zip file and decompress if haven't done yet\n",
+    "    data_path = os.path.join(data_folder, data_name)\n",
+    "\n",
+    "    if not os.path.exists(data_path):\n",
+    "        filename = 'ml-' + MOVIELENS_DATA_SIZE + '.zip'\n",
+    "        filepath = maybe_download('http://files.grouplens.org/datasets/movielens/'+filename, filename)\n",
+    "\n",
+    "        with ZipFile(filepath, 'r') as zf:\n",
+    "            zf.extractall()\n",
+    "    \n",
+    "        # remove zip file we already used\n",
+    "        os.remove(filepath)\n",
+    "    \n",
+    "    return data_path, separator        "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 59,
+   "metadata": {
+    "inputHidden": false,
+    "outputHidden": false
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div>\n",
+       "<style scoped>\n",
+       "    .dataframe tbody tr th:only-of-type {\n",
+       "        vertical-align: middle;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe tbody tr th {\n",
+       "        vertical-align: top;\n",
+       "    }\n",
+       "\n",
+       "    .dataframe thead th {\n",
+       "        text-align: right;\n",
+       "    }\n",
+       "</style>\n",
+       "<table border=\"1\" class=\"dataframe\">\n",
```

Comment:
This guy does not show properly....

Reply:
ok I fixed this as well, there was a misplaced $ sign ... I wonder why it worked in the Jupyter notebook :s